### ISSUE TITLE
Rewrite Electron front-end to get input directly from Whisper script rather than parsing vtt file

### ISSUE THREAD

**Issue by petersterne** [Stage-Whisper/Stage-Whisper#105]
Currently, the Electron app creates a transcription by parsing the vtt file output by the Whisper python script. Specifically, the Electron front-end waits until the Whisper script has finished transcribing the full audio file and written it to a vtt file, then converts the vtt file to json, parses it, splits it into an array of time-stamped lines, and then creates a transcription object that is stored in the sqlite transcription database.

Here's the relevant code: https://github.com/Stage-Whisper/Stage-Whisper/blob/use-backend/electron/electron/handlers/runWhisper/runWhisper.ts#L130

This behavior no longer makes sense for our use case, since @harrislapiroff has tweaked the Whisper script to output directly to stdio rather than producing a vtt file.

So we need to rewrite the Electron front-end so that it takes the Whisper lines directly from stdio (ideally as they're being written) as input. This should be a relatively easy fix to implement for anyone who has experience with Electron, React, or even plain JS. 

This is currently a program-breaking bug and fixing it is top priority. 

Although there is one obvious temporary workaround (just have the Whisper script produce a vtt file that the Electron front-end can parse, which is how it used to work), I believe it would be a poor solution for a number of reasons. 

First of all, it would add an unncessary step — writing the vtt file and then reading it, instead of just reading the transcription directly from stdio. Second, it would delay the Electron front-end, which would not be able to start writing transcribed lines to the database until the entire file has been transcribed. Finally, it would disrupt the program's overall architectural philosophy, which calls for the Electron front-end to manage all the file and database operations while the Whisper script only handles the actual transcription. Ensuring that each component of the program has a clearly-defined job is important for extensibility and future feature development (e.g. speaker diarization).

**Comment by sawhney17** [comment]
Could I be assigned this? This description you provided makes the blocker clear to me.

**Comment by petersterne** [comment]
Assigned, thanks @sawhney17!

**Comment by petersterne** [comment]
#107 implements the temporary workaround (having Whisper write a vtt file) to avoid breaking the program's functionality, but I expect this will be replaced within the next few days with a permanent solution.

**Comment by sawhney17** [comment]
Should I work on making it output to stdio now? Or work on making it read from stdio with the expectation that the backend will b rewritten soon? Is this still required? Sorry a little confused. 😅

**Comment by sawhney17** [comment]
looks like it's just a temporary workaround to convert to VTT so I guess I'll convert the run whisper script.

**Comment by petersterne** [comment]
Yeah, just comment out the "Save VTT" section (lines 89–92) the same way that the "Save TXT" section has been commented out.

That way, the backend won't produce any files, and all the transcription management (including export) will be handled by the Electron front-end.
